Actually check for strong authentication methods if the user specified that one
should be used.  If we don't, and the user actually specifies which strong
authentication method to use, we won't try it because we didn't check that the
server supports it, and just assumed that it doesn't.

--- fetchmail-6.2.0/pop3.c	2003-11-14 15:07:43.000000000 -0500
+++ fetchmail-6.2.0/pop3.c	2003-11-14 15:09:42.000000000 -0500
@@ -199,7 +199,7 @@
 	 * These authentication methods are blessed by RFC1734,
 	 * describing the POP3 AUTHentication command.
 	 */
-	if (ctl->server.authenticate == A_ANY)
+	if (ctl->server.authenticate != A_PASSWORD)
 	{
 	    if ((ok = capa_probe(sock)) != PS_SUCCESS)
 	    /* we are in STAGE_GETAUTH! */
